File: solutions/python/prime-factors/1/prime_factors.py
def factors(value):
    primes = []
    if value <=1:
        return primes

    num = 2
    while value > 1:
        while value % num == 0:
            primes.append(num)
            value //= num

        num += 1
        if num * num > value > 1:
            primes.append(value)
            break

    return  primes

File: solutions/cpp/prime-factors/1/prime_factors.cpp
#include "prime_factors.h"

namespace prime_factors {

    std::vector<long long>  of(long long x) {
        std::vector<long long> factors;
        long long num = 2;

        while (x > 1) {
            while ( x % num == 0 ) {
                factors.push_back(num);
                x /= num;
            }
            num++;
            if (num * num > x && x > 1) {
                factors.push_back(x);
                break;
            }
        }
        return factors;
    }

}  // namespace prime_factors
